### PR TITLE
add popData validation

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: src/vbk/pop_service_impl.cpp [hunk 1]
```diff
@@ -213,19 +213,61 @@ void PopServiceImpl::removePayloadsFromMempool(const altintegration::PopData& po
     mempool->removePayloads(popData);
 }
 
+bool popdataSizeValidation(const altintegration::PopData& popData, altintegration::ValidationState& state)
```
Comment (Warchant) [inline src/vbk/pop_service_impl.cpp]:
```suggestion
bool checkPopDataSize(const altintegration::PopData& popData, altintegration::ValidationState& state)
```

File: src/vbk/pop_service_impl.cpp [hunk 2]
```diff
@@ -213,19 +213,61 @@ void PopServiceImpl::removePayloadsFromMempool(const altintegration::PopData& po
...
+{
+    uint32_t nPopDataSize = ::GetSerializeSize(popData, CLIENT_VERSION);
```
Comment (Warchant) [inline src/vbk/pop_service_impl.cpp:223]: better use popData.estimateSize()

File: src/vbk/pop_service_impl.cpp [hunk 3]
```diff
@@ -213,19 +213,61 @@ void PopServiceImpl::removePayloadsFromMempool(const altintegration::PopData& po
...
+    auto& config = getService<Config>();
+
+    if (nPopDataSize >= config.popconfig.alt->getMaxPopDataSize()) {
+        return state.Invalid("popdataSizeValidation() fail: popData oversize", "popData raw size more than allowed");
```
Comment (Warchant) [inline src/vbk/pop_service_impl.cpp]:
```suggestion
        return state.Invalid("popdata-overisize", "popData raw size more than allowed");
```

File: src/vbk/pop_service_impl.cpp [hunk 4]
```diff
@@ -213,19 +213,61 @@ void PopServiceImpl::removePayloadsFromMempool(const altintegration::PopData& po
...
+    return true;
+}
+
+bool popdataStatelessValidation(const altintegration::PopData& popData, altintegration::ValidationState& state)
+{
+    auto& config = getService<Config>();
+
+    for (const auto& b : popData.context) {
+        if (!altintegration::checkBlock(b, state, *config.popconfig.vbk.params)) {
+            return state.Invalid("popdataStatelessValidation() failed", "vbk block invalid");
```
Comment (Warchant) [inline src/vbk/pop_service_impl.cpp]:
```suggestion
            return state.Invalid("pop-vbkblock-statelessly-invalid");
```

File: src/vbk/pop_service_impl.cpp [hunk 5]
```diff
@@ -213,19 +213,61 @@ void PopServiceImpl::removePayloadsFromMempool(const altintegration::PopData& po
...
+    return true;
+}
+
 bool addAllPayloadsToBlockImpl(altintegration::AltTree& tree, const CBlockIndex* indexPrev, const CBlock& block, BlockValidationState& state) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 {
     AssertLockHeld(cs_main);
 
+    altintegration::ValidationState instate;
+
+    if (!popdataSizeValidation(block.popData, instate) || !popdataStatelessValidation(block.popData, instate)) {
```
Comment (Warchant) [inline src/vbk/pop_service_impl.cpp]: move to ContextualCheckBlock

File: src/vbk/pop_service_impl.cpp (hunk 1 above)
Comment (Mr-Leshiy) [reply src/vbk/pop_service_impl.cpp]: fixed

File: src/vbk/pop_service_impl.cpp (hunk 3 above)
Comment (Mr-Leshiy) [reply src/vbk/pop_service_impl.cpp]: fixed

File: src/vbk/pop_service_impl.cpp (hunk 5 above)
Comment (Mr-Leshiy) [reply src/vbk/pop_service_impl.cpp]: fixed

**Review by Warchant — APPROVED** [review]

**Review by Warchant — APPROVED** [review]

**Pull request merged**